lib/generators/rails/templates/api_controller.rb

```diff
@@ -48,16 +48,28 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_<%= singular_table_name %>
+      <%- if Rails::VERSION::MAJOR >= 8 -%>
+      @<%= singular_table_name %> = <%= orm_class.find(class_name, "params.expect(:id)") %>
+      <%- else -%>
       @<%= singular_table_name %> = <%= orm_class.find(class_name, "params[:id]") %>
+      <%- end -%>
     end
 
     # Only allow a list of trusted parameters through.
     def <%= "#{singular_table_name}_params" %>
+      <%- if Rails::VERSION::MAJOR >= 8 -%>
+      <%- if attributes_names.empty? -%>
+      params.expect(<%= singular_table_name %>: {})
```

Comment:
If we replace `params.fetch` like this it changes the behavior. Now, even though there are no attributes expected, you must send at least one attribute or you'll get a 400 error.

I suggest changing only the case with attributes and leaving this one alone.

Reply:
@martinemde This makes sense, should the same be reverted from your PR on the Rails side?

https://github.com/rails/rails/pull/51674/files#diff-b20602dd9d67c64034125da9a3708746393dd155baa893f3bae0801dfb0efe11R51

and

https://github.com/rails/rails/pull/51674/files#diff-78fc500caeccee5a125dc41b391dec245fc90864a13ba44ef5ce009887438cd7R52

Reply:
You're exactly right. I submitted rails/rails#52932 to fix it. Thanks for noticing that I was in violation of my own recommendation. Thanks!

Reply:
This was the purpose of `params.allow`. Please someone from rails core let me know if that's worth a second, more focused attempt and I can extract the original code back out.